### PR TITLE
fix test cases

### PR DESCRIPTION
**Pull request by bishnuag**
- Below are a few test cases where latency/histogram is captured and are asserted 

1. test/endpoints/bar/bar_metrics_test.go
2. test/endpoints/baz/baz_metrics_test.go
3. test/endpoints/tchannel/baz/baz_metrics_test.go

**The metrics (latency/histogram) cannot be deterministic in nature, the histogram verifies as "0-10ms", or "10-25ms". These terms are added in keys and fail during asserts.** 

**_Examples -_** 

```
endpoint.latency-hist+apienvironment=production,bucket=10ms-25ms,bucketid=0002,clienttype=http,dc=unknown,device=ios,deviceversion=carbon,endpointid=bar,env=test,handlerid=normal,protocol=HTTP,regionname=san_francisco,service=test-gateway,status=200


{endpoint.latency-hist {COUNTER 1 0 0} 1665824496831231000 [{env test} {protocol HTTP} {device ios} {handlerid normal} {dc unknown} {clienttype http} {apienvironment production} {status 200} {deviceversion carbon} {endpointid bar} {service test-gateway} {regionname san_francisco} {bucketid 0002} {bucket 10ms-25ms}]} 
```
Here the term "bucket=10ms-25ms,bucketid=0002" in the key is non-deterministic & changes based on latency/histogram captured during runtime,leading to a test case failures.

**Action Items:**

1. Instead of taking the entire key, compare key prefixes
2. Updating this will not alter the meaning of the test case by any means

**Jira:** https://t3.uberinternal.com/browse/EDGE-12001

### CODE REVIEW

File: test/endpoints/bar/bar_metrics_test.go [hunk 1]
```diff
@@ -116,26 +116,27 @@ func TestCallMetrics(t *testing.T) {
...
-	assert.Contains(t, metrics, key, "expected metric: %s", key)
-	assert.Equal(t, int64(1), metrics[key].Value.Count)
+	assert.True(t, value > 1000, fmt.Sprintf("expected latency > 1000 nano seconds but found to be: %d", value))
+	assert.True(t, value < 50*1000*1000, fmt.Sprintf("expected latency < 50 milli seconds but found to be: %d", value))
+
+	key = "endpoint.latency-hist"
+	keyFound := false
+	for metricKeyName := range metrics {
+		if strings.Contains(metricKeyName, key) {
+			assert.Equal(t, int64(1), metrics[metricKeyName].Value.Count, fmt.Sprintf("key: %s, metric: %v\n", key, metrics[key]))
```
Comment (anirudhraja) [inline test/endpoints/bar/bar_metrics_test.go:135]: Should we add a nil check for metrics[metricKeyName] ?

Comment (bishnuag) [reply test/endpoints/bar/bar_metrics_test.go]: Addressed.

**Comment by anirudhraja** [comment]
LGTM

**Review by zeeshan-mohd — APPROVED** [review]
LGTM

**Pull request merged**